Job DEFAULT.metabase.task.abandonment-emails.job threw an unhandled Exception:

**Describe the bug**
Nightly when the Adandonment Email job runs, it causes an unhandled Exception to be thrown.

This has been occurring since Feb 06, 2020.  As best as I can tell from the logs, this is also when metabase was updated to version v0.34.1 (265695c release-0.34.x)

**Logs**
2020-03-06T01:00:01.595517293Z 03-06 12:00:01 INFO task.send-pulses :: Sending scheduled pulses...
2020-03-06T01:00:09.456048704Z 03-06 12:00:09 ERROR core.JobRunShell :: Job DEFAULT.metabase.task.abandonment-emails.job threw an unhandled Exception: 
2020-03-06T01:00:09.456092740Z clojure.lang.ExceptionInfo: Input to should-send-abandoment-email? does not match schema: 
2020-03-06T01:00:09.456100152Z 
2020-03-06T01:00:09.456106712Z 	   [nil (named (not (instance? java.time.temporal.Temporal a-metabase.models.user.UserInstance)) last-user) (named (not (instance? java.time.temporal.Temporal a-metabase.models.activity.ActivityInstance)) last-activity) (named (not (instance? java.time.temporal.Temporal a-metabase.models.view_log.ViewLogInstance)) last-view)]  
2020-03-06T01:00:09.456114313Z 
2020-03-06T01:00:09.456121957Z  {:type :schema.core/error, :schema [#schema.core.One{:schema (maybe java.time.temporal.Temporal), :optional? false, :name instance-creation} #schema.core.One{:schema (maybe java.time.temporal.Temporal), :optional? false, :name last-user} #schema.core.One{:schema (maybe java.time.temporal.Temporal), :optional? false, :name last-activity} #schema.core.One{:schema (maybe java.time.temporal.Temporal), :optional? false, :name last-view}], :value [(t/local-date-time "2016-08-03T12:00:21") #metabase.models.user.UserInstance{:last-user (t/local-date-time "2020-02-20T05:00:57")} #metabase.models.activity.ActivityInstance{:last-activity (t/local-date-time "2020-03-06T00:49:12")} #metabase.models.view_log.ViewLogInstance{:last-view (t/local-date-time "2020-03-06T00:49:10")}], :error [nil (named (not (instance? java.time.temporal.Temporal a-metabase.models.user.UserInstance)) last-user) (named (not (instance? java.time.temporal.Temporal a-metabase.models.activity.ActivityInstance)) last-activity) (named (not (instance? java.time.temporal.Temporal a-metabase.models.view_log.ViewLogInstance)) last-view)]}
2020-03-06T01:00:09.456152581Z 	at metabase.task.follow_up_emails$fn__71884$should_send_abandoment_email_QMARK___71893.invoke(follow_up_emails.clj:95)
2020-03-06T01:00:09.456158224Z 	at metabase.task.follow_up_emails$fn__71884$should_send_abandoment_email_QMARK___71893$fn__71894.invoke(follow_up_emails.clj:97)
2020-03-06T01:00:09.456163262Z 	at metabase.task.follow_up_emails$fn__71884$should_send_abandoment_email_QMARK___71893.invoke(follow_up_emails.clj:95)
2020-03-06T01:00:09.456176624Z 	at metabase.task.follow_up_emails$send_abandoment_email_if_needed_BANG_.invokeStatic(follow_up_emails.clj:119)
2020-03-06T01:00:09.456181904Z 	at metabase.task.follow_up_emails$send_abandoment_email_if_needed_BANG_.invoke(follow_up_emails.clj:114)
2020-03-06T01:00:09.456187257Z 	at metabase.task.follow_up_emails.AbandonmentEmail.execute(follow_up_emails.clj:135)
2020-03-06T01:00:09.456192265Z 	at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
2020-03-06T01:00:09.456197303Z 	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:557)
2020-03-06T01:00:09.491373675Z 03-06 12:00:09 ERROR core.ErrorLogger :: Job (DEFAULT.metabase.task.abandonment-emails.job threw an exception.
2020-03-06T01:00:09.491415762Z org.quartz.SchedulerException: Job threw an unhandled exception. [See nested exception: clojure.lang.ExceptionInfo: Input to should-send-abandoment-email? does not match schema: 
2020-03-06T01:00:09.491428645Z 	   [nil (named (not (instance? java.time.temporal.Temporal a-metabase.models.user.UserInstance)) last-user) (named (not (instance? java.time.temporal.Temporal a-metabase.models.activity.ActivityInstance)) last-activity) (named (not (instance? java.time.temporal.Temporal a-metabase.models.view_log.ViewLogInstance)) last-view)]  
2020-03-06T01:00:09.491452378Z 
2020-03-06T01:00:09.491457603Z  {:type :schema.core/error, :schema [#schema.core.One{:schema (maybe java.time.temporal.Temporal), :optional? false, :name instance-creation} #schema.core.One{:schema (maybe java.time.temporal.Temporal), :optional? false, :name last-user} #schema.core.One{:schema (maybe java.time.temporal.Temporal), :optional? false, :name last-activity} #schema.core.One{:schema (maybe java.time.temporal.Temporal), :optional? false, :name last-view}], :value [(t/local-date-time "2016-08-03T12:00:21") #metabase.models.user.UserInstance{:last-user (t/local-date-time "2020-02-20T05:00:57")} #metabase.models.activity.ActivityInstance{:last-activity (t/local-date-time "2020-03-06T00:49:12")} #metabase.models.view_log.ViewLogInstance{:last-view (t/local-date-time "2020-03-06T00:49:10")}], :error [nil (named (not (instance? java.time.temporal.Temporal a-metabase.models.user.UserInstance)) last-user) (named (not (instance? java.time.temporal.Temporal a-metabase.models.activity.ActivityInstance)) last-activity) (named (not (instance? java.time.temporal.Temporal a-metabase.models.view_log.ViewLogInstance)) last-view)]}]
2020-03-06T01:00:09.491481582Z 	at org.quartz.core.JobRunShell.run(JobRunShell.java:224)
2020-03-06T01:00:09.491486805Z 	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:557)
2020-03-06T01:00:09.491491837Z Caused by: clojure.lang.ExceptionInfo: Input to should-send-abandoment-email? does not match schema: 
2020-03-06T01:00:09.491497126Z 
2020-03-06T01:00:09.491502923Z 	   [nil (named (not (instance? java.time.temporal.Temporal a-metabase.models.user.UserInstance)) last-user) (named (not (instance? java.time.temporal.Temporal a-metabase.models.activity.ActivityInstance)) last-activity) (named (not (instance? java.time.temporal.Temporal a-metabase.models.view_log.ViewLogInstance)) last-view)]  
2020-03-06T01:00:09.491509291Z 
2020-03-06T01:00:09.491529665Z  {:type :schema.core/error, :schema [#schema.core.One{:schema (maybe java.time.temporal.Temporal), :optional? false, :name instance-creation} #schema.core.One{:schema (maybe java.time.temporal.Temporal), :optional? false, :name last-user} #schema.core.One{:schema (maybe java.time.temporal.Temporal), :optional? false, :name last-activity} #schema.core.One{:schema (maybe java.time.temporal.Temporal), :optional? false, :name last-view}], :value [(t/local-date-time "2016-08-03T12:00:21") #metabase.models.user.UserInstance{:last-user (t/local-date-time "2020-02-20T05:00:57")} #metabase.models.activity.ActivityInstance{:last-activity (t/local-date-time "2020-03-06T00:49:12")} #metabase.models.view_log.ViewLogInstance{:last-view (t/local-date-time "2020-03-06T00:49:10")}], :error [nil (named (not (instance? java.time.temporal.Temporal a-metabase.models.user.UserInstance)) last-user) (named (not (instance? java.time.temporal.Temporal a-metabase.models.activity.ActivityInstance)) last-activity) (named (not (instance? java.time.temporal.Temporal a-metabase.models.view_log.ViewLogInstance)) last-view)]}
2020-03-06T01:00:09.491539563Z 	at metabase.task.follow_up_emails$fn__71884$should_send_abandoment_email_QMARK___71893.invoke(follow_up_emails.clj:95)
2020-03-06T01:00:09.491544616Z 	at metabase.task.follow_up_emails$fn__71884$should_send_abandoment_email_QMARK___71893$fn__71894.invoke(follow_up_emails.clj:97)
2020-03-06T01:00:09.491550334Z 	at metabase.task.follow_up_emails$fn__71884$should_send_abandoment_email_QMARK___71893.invoke(follow_up_emails.clj:95)
2020-03-06T01:00:09.491555003Z 	at metabase.task.follow_up_emails$send_abandoment_email_if_needed_BANG_.invokeStatic(follow_up_emails.clj:119)
2020-03-06T01:00:09.491560025Z 	at metabase.task.follow_up_emails$send_abandoment_email_if_needed_BANG_.invoke(follow_up_emails.clj:114)
2020-03-06T01:00:09.491565799Z 	at metabase.task.follow_up_emails.AbandonmentEmail.execute(follow_up_emails.clj:135)
2020-03-06T01:00:09.491571236Z 	at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
2020-03-06T01:00:09.491576195Z 	... 1 more

**To Reproduce**
Steps to reproduce the behavior:
1. Configure SMTP Details
2. Enable anonymous usage tracking
3. Wait for AbandonmentEmail job to run at midday.
4. See error in logs.

**Expected behavior**
The exception should not be occurring.

**Information about your Metabase Installation:**
    "version": {
      "date": "2020-02-25",
      "tag": "v0.34.3",
      "branch": "release-0.34.x",
      "hash": "1a83edb"
    },

**Severity**
This is an annoying Exception in the logs.  I'm currently trying to better monitor metabase for issues in the installation and this exception is an false issue.

--- a/src/metabase/core.clj
+++ b/src/metabase/core.clj
@@ -39,6 +39,7 @@
         (str (deferred-trs "Metabase Enterprise Edition extensions are PRESENT.")
              "\n\n"
              (deferred-trs "Usage of Metabase Enterprise Edition features are subject to the Metabase Commercial License.")
+             " "
              (deferred-trs "See {0} for details." "https://www.metabase.com/license/commercial/"))
         (deferred-trs "Metabase Enterprise Edition extensions are NOT PRESENT."))))
 

--- a/src/metabase/task/follow_up_emails.clj
+++ b/src/metabase/task/follow_up_emails.clj
@@ -90,11 +90,17 @@
 
 (s/defn ^:private should-send-abandoment-email?
   ([]
-   (should-send-abandoment-email?
-    (instance-creation-timestamp)
-    (db/select-one [User [:%max.date_joined :last-user]])
-    (db/select-one [Activity [:%max.timestamp :last-activity]])
-    (db/select-one [ViewLog [:%max.timestamp :last-view]])))
+   (let [[last-user last-activity last-view]
+         (map :last (db/query
+                     {:union-all
+                      [{:select [[:%max.date_joined :last]], :from [User]}
+                       {:select [[:%max.timestamp :last]], :from [Activity]}
+                       {:select [[:%max.timestamp :last]], :from [ViewLog]}]}))]
+     (should-send-abandoment-email?
+      (instance-creation-timestamp)
+      last-user
+      last-activity
+      last-view)))
 
   ([instance-creation :- (s/maybe Temporal)
     last-user         :- (s/maybe Temporal)

--- a/test/metabase/task/follow_up_emails_test.clj
+++ b/test/metabase/task/follow_up_emails_test.clj
@@ -20,6 +20,9 @@
              (-> @inbox vals first count))))))
 
 (deftest should-send-abandoment-email-test
+  (testing "Make sure zero-arity version works (#12068)"
+    (testing `(follow-up-emails/should-send-abandoment-email?)
+      (is (boolean? (#'follow-up-emails/should-send-abandoment-email?)))))
   (testing "Conditions where abandonment emails should be sent"
     (doseq [now               [(t/zoned-date-time) (t/offset-date-time) (t/instant)]
             instance-creation [0 1 5 nil]

